### PR TITLE
E2e test

### PR DESCRIPTION
**Pull request by dariajung**
@rkirov The feature for determining output filepath/file write is subject to change, so review commits from August 12 which deal with E2E testing `start setting up e2e testing` and `WIP e2e`.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -41,8 +42,15 @@ gulp.task('unit.test', ['test.compile'], function() {
   return gulp.src('build/test/unit/main_test.js', {read: false}).pipe(mocha(mochaOptions));
 });
 
+gulp.task('e2e.test', ['test.compile'], function() {
+  exec('node ./build/test/e2e/small_e2e.js', function(err, stdout, stderr) {
+    if (stdout) console.log(stdout);
+    if (stderr) console.log(stderr);
```
Comment (rkirov) [inline gulpfile.js]: print `err` too

File: gulpfile.js [hunk 2]
```diff
@@ -41,8 +42,15 @@ gulp.task('unit.test', ['test.compile'], function() {
   return gulp.src('build/test/unit/main_test.js', {read: false}).pipe(mocha(mochaOptions));
 });
 
+gulp.task('e2e.test', ['test.compile'], function() {
+  exec('node ./build/test/e2e/small_e2e.js', function(err, stdout, stderr) {
```
Comment (rkirov) [inline gulpfile.js]: why exec here, and not just `require('small_e2e.js')`?

File: test/e2e/small_e2e.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+                         '../../build/output');
+  // compile renamed program
+  var child = exec('tsc', function(error, stdout, stderr) {
+    if (stdout) console.log(stdout);
+    if (stderr) console.log(stderr);
+    if (error !== null) {
+      console.log('exec error: ' + error);
+    }
+  });
+  require('../../output/test/input/e2e_input');
```
Comment (rkirov) [inline test/e2e/small_e2e.ts]: add a comment here, 'execute the renamed and compiled program'

**Comment by rkirov** [comment]
LGTM, but try to remove `exec('node ...')` call.

File: gulpfile.js (hunk 2 above)
Comment (dariajung) [inline gulpfile.js]: changed to `require`

**Comment by rkirov** [comment]
looks great, push it!

**Pull request closed without merging**